spec/features/routing/full_scenario_spec.rb

```diff
@@ -165,7 +174,25 @@ def register_instructeur_and_log_in(email)
     expect(page).to have_content 'Mot de passe enregistré'
   end
 
-  def log_out
-    click_on 'Se déconnecter'
+  def log_out(old_layout: false)
+    if old_layout
+      wait_until { page.has_content?('Se déconnecter') }
+      click_on 'Se déconnecter'
+    else
+      try_2_times do
+        wait_until { page.has_css?('[title="Mon compte"]') }
+        find('[title="Mon compte"]').click
+        wait_until { page.has_content?('Se déconnecter') }
+        click_on 'Se déconnecter'
```

Comment:
En théorie y'aurait moyen d'utiliser les helpers de Capybara qui attendent automatiquement, pour éviter les faux positifs de la même manière.

Ici pour simplifier le matching je proposerais bien d'améliorer le markup du dropdown "Mon compte" en rajoutant un `aria-label` :

```haml
  %button.button.dropdown-button.header-menu-button{ 'aria-label': "Mon compte" }
    = image_tag "icons/account-circle.svg", title: "Mon compte"
```

Ce qui permet ensuite d'écrire une séquence plus simple que d'utiliser `wait_until` :

```ruby
click_button 'Mon compte'     # La méthode d'action va attendre qu'un élément matchant soit visible.
                              # "Mon compte" devrait être suffisant pour matcher le aria-label (et sinon [aria-label="Mon compte"] )
click_link 'Se déconnecter'   # Pareil, la méthode d'action va attendre qu'un élément matchant soit visible.
```

Reply:
- je viens d'essayer de remplacer le wait_until par `expect(page).to have_css('[title="Mon compte"]')` en retirant également le `try_2_times` je me prends un `ElementNotInteractableError`.

Reply:
et pour le `aria-label` ca ne marche pas mieux ` Capybara::ElementNotFound: Unable to find link or button "Mon compte"`